test/quantization/test_quant_api.py

```diff
@@ -423,20 +430,31 @@ def get_per_token_block_size(x):
         # input settings
         input_mapping_type = MappingType.ASYMMETRIC
         input_target_dtype = torch.int8
-        input_quant_func = lambda x: AffineQuantizedTensor.from_float(x, input_mapping_type, get_per_token_block_size(x), input_target_dtype)
-
-        def dynamic_quant(linear):
-            # note: order is important
-            linear.weight = torch.nn.Parameter(AffineQuantizedTensor.from_float(linear.weight, mapping_type, block_size, target_dtype, quant_min, quant_max, eps), requires_grad=False)
-            linear.weight = torch.nn.Parameter(LinearActQuantizedTensor.from_float(linear.weight, input_quant_func), requires_grad=False)
+        input_quant_func = lambda x: to_aqt(x, input_mapping_type, get_per_token_block_size(x), input_target_dtype)
 
         m = ToyLinearModel().eval()
         m_copy = copy.deepcopy(m)
         example_inputs = m.example_inputs()
-        dynamic_quant(m.linear1)
-        dynamic_quant(m.linear2)
+
+        weight_quantizer = TensorSubclassQuantizer(
+            to_aqt,
```

Comment:
``
def my_to_aqt(inp):
    to_aqt(inp, mapping_type, ...)
```

Can I pass `my_to_aqt` instead of `to_aqt`? If so, then this could be generic.

Reply:
`TensorSubclassQuantizer` is already generic enough to do this:

```
some_quantizer = TensorSubclassQuantizer(to_my_dtype, kwarg1=value1, kwarg2=value2, ...)
```

Reply:
Hm, but then `TensorSubclassQuantizer` needs to support `*args` and `**kwargs` if I want to pass something. Why not pass a curried function? Sure, it's a bit of a nit, but why not do 

```
TensorSubclassQuantizer(lambda input: to_my_dtype(input, kwarg1=value1, kwarg2=value2, ...))
```

Also, does `TensorSubclassQuantizer` only target `F.linear`? It seems like it's mostly a wrapper around `_replace_with_custom_fn_if_matches_filter`. Why not just use that function directly?

Reply:
OK sure, I think passing a curried function seems OK, it's just slightly harder for users probably

so `TensorSubclassQuantizer` is supposed to be a model level quantizer, we can expand this to support other things, but right now it's only for linear. I don't feel we want people to use `_replace_with_custom_fn_if_matches_filter` as a top level public api

Reply:
@jerryzh168 - Maybe not as a top-level API, but I thought this is first about unifying and replacing the code that we use to implement our toplevel `apply_dynamic_quant`, etc.

Reply:
we can deprecate `apply_dynamic_quant` as well I think, that one is a very specific configuration of a general quantizer